### PR TITLE
@ # 502 | use remote options

### PR DESCRIPTION
**Pull request by datphan**
connect #502

### CODE REVIEW

**Review by phuonglm — DISMISSED** [review]
still having problem with this bug #516 but we can fix later

File: lib/teracy-dev/loader.rb [hunk 1]
```diff
@@ -74,17 +74,21 @@ def sync_teracy_dev_entry(location)
...
       # this is useful to init the teracy-dev-entry or to override existing settings to enable auto sync
       # TERACY_DEV_ENTRY_LOCATION_GIT, TERACY_DEV_ENTRY_LOCATION_BRANCH
       # TERACY_DEV_ENTRY_LOCATION_REF, TERACY_DEV_ENTRY_LOCATION_SYNC
-      location['git'] = ENV['TERACY_DEV_ENTRY_LOCATION_GIT'] if ENV['TERACY_DEV_ENTRY_LOCATION_GIT']
-      location['branch'] = ENV['TERACY_DEV_ENTRY_LOCATION_BRANCH'] if ENV['TERACY_DEV_ENTRY_LOCATION_BRANCH']
-      location['tag'] = ENV['TERACY_DEV_ENTRY_LOCATION_TAG'] if ENV['TERACY_DEV_ENTRY_LOCATION_TAG']
-      location['ref'] = ENV['TERACY_DEV_ENTRY_LOCATION_REF'] if ENV['TERACY_DEV_ENTRY_LOCATION_REF']
+      location["git"] = {
+        "remote" => {
+          "origin" => ENV['TERACY_DEV_ENTRY_LOCATION_GIT']
```
Comment (hoatle) [inline lib/teracy-dev/loader.rb]: deprecate the old env vars and use the env var specififed here instead https://github.com/teracyhq/dev/issues/502#issuecomment-430094691 to match the new config format

File: lib/teracy-dev/location/git_synch.rb [hunk 2]
```diff
@@ -13,32 +14,53 @@ def sync(location, sync_existing)
...
-        ref = location['ref']
         lookup_path = location['lookup_path']
+
         path = location['path']
-        dir = location['dir']
+
+        git_config = location['git']
+
+        if git_config.instance_of? String
+          @logger.warn("Deprecated string value at location.git of #{path}, please use location.git.remote.origin instead")
```
Comment (hoatle) [inline lib/teracy-dev/location/git_synch.rb]: make the message easier to track where is the problem is:

```ruby
          @logger.warn("Deprecated string value at location.git of location: #{location}, please use location.git.remote.origin instead")
```

**Review by hoatle — CHANGES_REQUESTED** [review]
@datphan pls check my comments

**Comment by datphan** [comment]
I've just updated it, please check again

**Review by hoatle — APPROVED** [review]
LGTM

**Pull request merged**